lib/ui/screen/main.dart

```diff
@@ -215,26 +219,20 @@ class _MyHomePageState extends State<MyHomePage> {
                               alertProvider.setOrder(order);
                             },
                           ),
+
                           Expanded(
-                            child: ListView.builder(
-                              itemBuilder: (context, position) {
-                                return Column(
-                                    mainAxisAlignment: MainAxisAlignment.center,
-                                    crossAxisAlignment:
-                                        CrossAxisAlignment.center,
-                                    children: [
-                                      buildSlidable(
-                                          context,
-                                          alertProvider.items[position],
-                                          alertProvider),
-                                      Divider(
-                                        color: Theme.of(context)
-                                            .colorScheme
-                                            .background,
-                                      )
-                                    ]);
+                            child: AnimatedList(
+                              key: alertProvider.listKey,
+                              initialItemCount: alertProvider.items.length,
+                              itemBuilder: (context, position, animation) {
+                                if (position < alertProvider.items.length) {
```

Comment:
It's weird that I have to check the position < size to prevent index out of bound...

Reply:
without the check, will we run into index out of bound? I thought the position passed in is always legal position.